### DISCUSSION TITLE
User-supplied pseudo-random number generator seed

### DISCUSSION THREAD

**Discussion by wlandau** [quarto-dev/quarto-cli#2911] [Q&A]
Would it be feasible to let users supply a seed as a `quarto render` flag? Would it work for all language engines?

**Comment by cscheid** [comment]
That's unlikely to be near-term practical for precisely the reason you've pointed out. There's also issues of ordering and consistency under creation of new documents: For example, would you want the PRNG to be reset from .qmd to .qmd? There's reasons to want to do it and reasons to want to not do it.

Random-number generation is a _very_ tricky thing to get right. I don't think it's the kind of technology that we can define sensible defaults that "just work" across languages and deployments, and I'd rather that quarto not make that mistake.
